backend/workflow_manager/workflow_v2/views.py

```diff
@@ -48,7 +60,12 @@ def make_execution_response(response: ExecutionResponse) -> Any:
 
 class WorkflowViewSet(viewsets.ModelViewSet):
     versioning_class = URLPathVersioning
-    permission_classes = [IsOwner]
+
+    def get_permissions(self) -> list[Any]:
+        if self.action in ["destroy", "partial_update"]:
```

Comment:
@johnyrahul shouldn't we also add update to this list to ensure shared users can't edit with a PUT call?
```suggestion
        if self.action in ["destroy", "partial_update", "update"]:
```